### DISCUSSION TITLE
How to get notified on job success/failure or check running job status?

### DISCUSSION THREAD

**Discussion by derekleverenz** [apalis-dev/apalis#319] [Q&A] [answered]
I'm working on system where some jobs depend on other jobs, I want to queue one or more jobs, wait for them to complete, then queue one or more other jobs in response. I'm using the postgres storage as well.

Is there a supported way to check the status of a queued/running job? I was planning to build a simple task that would poll for the job, check the status, and sleep if it was still pending, but I can't find anywhere in the API that will let me check the status of a job. Storage::fetch_by_id looks like what I want but I don't see how to actually get the status of the job from the returned Request or Job, maybe I'm just missing it.

Alternatively I can always just query the apalis.jobs table directly and check the status column, but I was hoping to find a way supported in the programmatic API.

Thanks!

**Comment by geofmureithi** [comment]
I think this is a case where you might want to do the query yourself.
You can use a tower Layer to standardize this.
The `AckLayer` is a good example of how to do it and reduce the number of calls made to the db.

**Reply by derekleverenz** [reply]
thanks ill take a look at that too. The mistake I made was also just looking at the docs for apalis and not the other crates

**Comment by geofmureithi** [comment, marked as answer]
Also you can use `fetch_by_id`. It will return `Request` which had a data entry for `SqlContext`. 
```rs
let ctx: SqlContext = req.data().unwrap():
```
The context contains a state which tells you the state of the job.

**Reply by derekleverenz** [reply]
got it thanks! this is what i was looking for but you are right that I'll want some custom solution anyway, I started working on something for that so I only have one task doing polling and making database calls
